lib/domain/repository/interfaces/i_movies_repository.dart

```diff
@@ -1,7 +1,6 @@
-import '../../entity/movie.dart';
+import '../../../core/utils/resources/data_state.dart';
+import '../../../data/model/result_model.dart';
 
 abstract class IMoviesRepository {
-  Future<List<Movie>> getMovies();
-
-  Future<Movie> getMovieById(int id);
+  Future<DataState<ResultModel>> fetchMovies(chosenEndpoint);
```

Comment:
What should be the type of `chosenEndpoint?`

Reply:
It's an Endpoint, i forgot to write the type jeje. Fixed it!